File: 2025/day03/prob1.py
def find(l):
    m = 0
    index = 0
    for i in range(len(l)):
        if l[i] > m:
            m = l[i]
            index = i
    return (m, index)


def main():
    with open("2025/day03/input.txt") as f:
        banks = f.read().split("\n")
    p = 0
    for bank in banks:
        bank = [int(n) for n in list(bank)]
        m, index = find(bank[:-1])
        n = max(bank[index+1:])
        p += 10*m + n
    print(p)
    

if __name__ == "__main__":
    main()

File: 2025/day03/Prob1.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class Prob1 {

    public static void main(String[] args) throws Exception {
        List<String> banks = Files.readAllLines(Path.of("2025/day03/input.txt"));
        int n = 0;
        int n1;
        int n2;
        int i_max;
        int j_max;
        for (String bank: banks) {
            i_max = 0;
            for (int i = 0; i < bank.length() - 1; i++) {
                if (bank.charAt(i) > bank.charAt(i_max)) {
                    i_max = i;
                }
            }
            n1 = bank.charAt(i_max) - '0';
            j_max = i_max + 1;
            for (int j = i_max + 1; j < bank.length(); j++) {
                if (bank.charAt(j) > bank.charAt(j_max)) {
                    j_max = j;
                }
            }
            n2 = bank.charAt(j_max) - '0';
            n += 10*n1 + n2;
        }
        System.out.println(n);
    }

}